src/compiler/core.ts

```diff
@@ -1773,9 +1773,27 @@ namespace ts {
         if (a === b) return Comparison.EqualTo;
         if (a === undefined) return Comparison.LessThan;
         if (b === undefined) return Comparison.GreaterThan;
-        a = a.toUpperCase();
-        b = b.toUpperCase();
-        return a < b ? Comparison.LessThan : a > b ? Comparison.GreaterThan : Comparison.EqualTo;
+        const aLength = a.length;
+        const bLength = b.length;
+        const maxLength = Math.min(aLength, bLength);
+        for (let i = 0; i < maxLength; i++) {

```

Comment:
One of the perks of operating this way is that you could extend the function to take in the start and end indices of each string as arguments, such that places where you do a substring comparison can avoid the substring allocation. Not sure how much it matters in practice, though.